File: backend/app/api/v1/routes/legal.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.dependencies import get_db
from app.schemas.legal import LegalQARequest, LegalQAResponse
from app.services.legal_service import LegalService

router = APIRouter()


@router.post("/qa", response_model=LegalQAResponse)
async def legal_qa(
    request: LegalQARequest,
    db: AsyncSession = Depends(get_db),
) -> LegalQAResponse:
    """Ask a legal question and receive an AI-assisted answer."""
    service = LegalService(db)
    return await service.answer_question(request)


@router.get("/qa/history")
async def get_qa_history(
    limit: int = Query(20, ge=1, le=100),
    db: AsyncSession = Depends(get_db),
) -> list[dict]:
    """Get recent legal Q&A history."""
    service = LegalService(db)
    return await service.get_history(limit)

File: frontend/src/lib/api/legal.ts
import { apiClient } from "./client";
import type { ApiResponse, PaginatedResponse } from "@/types/api";
import type {
  LegalQARequest,
  LegalQAResponse,
  LegalQAHistoryItem,
} from "@/types/legal";

export async function askQuestion(
  request: LegalQARequest
): Promise<ApiResponse<LegalQAResponse>> {
  return apiClient.post<ApiResponse<LegalQAResponse>>("/legal/ask", request);
}

export async function getHistory(
  page = 1,
  perPage = 20
): Promise<PaginatedResponse<LegalQAHistoryItem>> {
  return apiClient.get<PaginatedResponse<LegalQAHistoryItem>>(
    "/legal/history",
    {
      page: page.toString(),
      per_page: perPage.toString(),
    }
  );
}
